packages/flutter/lib/src/services/scribe.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'message_codec.dart';
+import 'system_channels.dart';
+
+/// An interface into Android's stylus handwriting text input.
+///
+/// Allows handwriting directly on top of a text input using a stylus.
+///
+/// See also:
+///
+///  * [EditableText.stylusHandwritingEnabled], which controls whether Flutter's
+///    built-in text fields support handwriting input.
+///  * [SystemChannels.scribe], which is the [MethodChannel] used by this
+///    class, and which has a list of the methods that this class handles.
+final class Scribe {
+  static const MethodChannel _channel = SystemChannels.scribe;
+
+  /// A convenience method to check if the device currently supports Scribe
+  /// stylus handwriting input.
+  ///
+  /// Call this each time before calling [startStylusHandwriting] to make sure
+  /// it's available.
+  ///
+  /// The return type is nullable due to the way
+  /// [MethodChannel.invokeMethod](https://main-api.flutter.dev/flutter/services/MethodChannel/invokeMethod.html)
+  /// works, but a successful call will never resolve to `null`.
+  ///
+  /// {@tool snippet}
+  /// This example shows using [isFeatureAvailable] to confirm that
+  /// [startStylusHandwriting] can be called.
+  ///
+  /// ```dart
+  /// if (!(await Scribe.isFeatureAvailable() ?? false)) {
+  ///   // The device doesn't support stylus input right now, or maybe at all.
+  ///   return;
+  /// }
+  ///
+  /// // Scribe is supported, so start it.
+  /// Scribe.startStylusHandwriting();
+  /// ```
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  /// * [isStylusHandwritingAvailable], which is similar, but throws an error
+  ///   when called by an unsupported API level. It directly corresponds to the
+  ///   underlying Android API
+  ///   <https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#isStylusHandwritingAvailable()>.
+  ///  * [EditableText.stylusHandwritingEnabled], which controls whether
+  ///    Flutter's built-in text fields support handwriting input.
+  static Future<bool?> isFeatureAvailable() {
+    return _channel.invokeMethod<bool?>(
+      'Scribe.isFeatureAvailable',
+    );
+  }
+
+  /// Returns true if the InputMethodManager supports Scribe stylus handwriting
+  /// input, false otherwise.
+  ///
+  /// Call this each time before calling [startStylusHandwriting] to make sure
```

Comment:
If this is the case shouldn't this API be incorporated into `startStylusHandwriting` then? Since they are always paired?

Reply:
It sounds like it should, but that's not the way that Android designed their API for whatever reason. I want to provide an accurate reproduction of the Android API in case our users need it.

Reply:
If we want to 100% match the Android API should we call this class`AndroidScribe` then? Also are there any android handwriting overview in their docs we could link in the class API docs?

Reply:
I just made an attempt at renaming this but got halfway through and started having second thoughts. Following this naming, Scribble should be called iOSScribble, and the method channel names don't quite match (`Scribe.x` instead of `AndroidScribe.x`).

I'll make sure the docs are clear about the direct connection to Android's APIs. I found a good page to link to like you suggested: https://developer.android.com/develop/ui/views/touch-and-input/stylus-input/stylus-input-in-text-fields